compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/ClassReference.kt

```diff
@@ -93,38 +94,41 @@ public fun ClassReference.allSuperTypeClassReferences(
   includeSelf: Boolean = false
 ): Sequence<ClassReference> {
 
-  fun KtClassOrObject.superTypeEntryNames() = superTypeListEntries
-    .map { it.requireFqName(module) }
-
-  fun List<FqName>.typeRefs(): Sequence<ClassReference> {
-    return asSequence().map { module.requireClassReference(it) }
+  val initial = if (includeSelf) {
+    sequenceOf(this)
+  } else {
+    directSuperClassReferences(module)
   }
 
-  val initial = sequenceOf(this).drop(if (includeSelf) 0 else 1)
-
   return generateSequence(initial) { superTypes ->
     superTypes
-      .mapNotNull { classRef ->
-        when (classRef) {
-          is Psi -> {
-            classRef.clazz.superTypeEntryNames()
-              .takeIf { it.isNotEmpty() }
-              ?.typeRefs()
-          }
-          is Descriptor -> {
-            classRef.clazz.directSuperClassAndInterfaces()
-              .takeIf { it.isNotEmpty() }
-              ?.asSequence()
-              ?.map { module.requireClassReference(it.fqNameSafe) }
-          }
-        }
-      }
+      .mapNotNull { classRef -> classRef.directSuperClassReferences(module) }
       .flatten()
       .takeIf { it.firstOrNull() != null }
   }.flatten()
     .distinctBy { it.fqName }
 }
 
+@ExperimentalAnvilApi
+public fun ClassReference.directSuperClassReferences(
+  module: ModuleDescriptor
+): Sequence<ClassReference> {
+
+  return when (this) {
+    is Descriptor -> {
+      clazz.directSuperClassAndInterfaces()
+        .asSequence()
+        .map { module.requireClassReference(it.fqNameSafe) }
+    }
+    is Psi -> {
+      clazz.superTypeListEntries
+        .map { it.requireFqName(module) }
+        .asSequence()
```

Comment:
Move the `asSequence()` operator one line up.
```suggestion
        .asSequence()
        .map { it.requireFqName(module) }
```